store/searchlayer/channel_layer.go

```diff
@@ -148,12 +149,17 @@ func (c *SearchChannelStore) AutocompleteInTeam(teamId string, term string, incl
 
 	if allFailed {
 		mlog.Debug("Using database search because no other search engine is available")
-		channelList, err = c.ChannelStore.AutocompleteInTeam(teamId, term, includeDeleted)
-		if err != nil {
-			return nil, err
+		channelList, nErr = c.ChannelStore.AutocompleteInTeam(teamId, term, includeDeleted)
+		if nErr != nil {
+			return nil, model.NewAppError("AutocompleteInTeam", "app.channel.search.app_error", nil, nErr.Error(), http.StatusInternalServerError)
 		}
 	}
-	return channelList, err
+
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	return channelList, nil
```

Comment:
This changes behavior slightly. Have you ensured that `channelList` is always nil in case of an error. If so, we don't need to change it anyways.

Reply:
Ok maybe you could be doing something with `channelList` even when you have an error, in that case I will change the `nil` by `channelList`.

By the way this change is to avoid the `<nil value but not nil pointer interface> != nil = true` and all that stuff you know :slightly_smiling_face:

Reply:
> By the way this change is to avoid the <nil value but not nil pointer interface> != nil = true and all that stuff you know slightly_smiling_face

Yes, exactly. Thank you for keeping that in mind! Your new changes look good to me!